src/backend/screencast.rs

```diff
@@ -0,0 +1,241 @@
+        options: CreateSessionOptions,
+    ) -> Result<CreateSessionResponse>;
+
+    async fn session_closed(&self, session_handle: OwnedObjectPath) -> Result<()>;
+
+    async fn select_sources(
+        &self,
+        session_handle: OwnedObjectPath,
+        app_id: Option<AppID>,
+        options: SelectSourcesOptions,
+    ) -> Result<SelectSourcesResponse>;
+
+    async fn start_cast(
+        &self,
+        session_handle: OwnedObjectPath,
+        app_id: Option<AppID>,
+        window_identifier: Option<WindowIdentifierType>,
+        options: StartCastOptions,
+    ) -> Result<StartCastResponse>;
+}
+
+pub(crate) struct ScreencastInterface {
+    imp: Arc<dyn ScreencastImpl>,
+    cnx: zbus::Connection,
+}
+
+impl ScreencastInterface {
+    pub fn new(imp: Arc<dyn ScreencastImpl>, cnx: zbus::Connection) -> Self {
+        Self { imp, cnx }
+    }
+}
+
+#[zbus::interface(name = "org.freedesktop.impl.portal.ScreenCast")]
+impl ScreencastInterface {
+    #[zbus(
+        property(emits_changed_signal = "const"),
+        name = "AvailableSourceTypes"
+    )]
+    fn available_source_types(&self) -> u32 {
+        let imp = Arc::clone(&self.imp);
+        imp.available_source_types().bits()
+    }
+
+    #[zbus(
+        property(emits_changed_signal = "const"),
+        name = "AvailableCursorModes"
+    )]
+    fn available_cursor_mode(&self) -> u32 {
+        let imp = Arc::clone(&self.imp);
+        imp.available_cursor_mode().bits()
+    }
+
+    #[zbus(property(emits_changed_signal = "const"), name = "version")]
+    fn version(&self) -> u32 {
+        5
+    }
+
+    #[zbus(name = "CreateSession")]
+    #[zbus(out_args("response", "results"))]
+    async fn create_session(
+        &self,
+        handle: OwnedObjectPath,
+        session_handle: OwnedObjectPath,
+        app_id: MaybeAppID,
+        options: CreateSessionOptions,
+    ) -> Result<Response<CreateSessionResponse>> {
+        let imp = Arc::clone(&self.imp);
+
+        #[cfg(feature = "tracing")]
+        tracing::debug!("IN ScreencastInterface::create_session(): {handle}, {session_handle}, …");
+
+        Request::spawn(
+            "ScreenCast::CreateSession",
+            &self.cnx,
+            handle.clone(),
+            Arc::clone(&self.imp),
+            async move {
+                let session = Session::new(session_handle);
+                let path = ObjectPath::from_string_unchecked(session.path().as_str().to_string());
+                let response = imp
+                    .create_session(handle, path.into(), app_id.inner(), options)
+                    .await;
+                if response.is_ok() {
+                    #[cfg(feature = "tracing")]
+                    tracing::debug!("session created: {:?}", session);
+                    let path = session.path().as_str().to_string();
+                    self.cnx.object_server().at(path.clone(), session).await?;
+                    if let Some(name) = self.cnx.unique_name() {
```

Comment:
The unique name should be retrieved from the handle token. if needed, we can add a method there for that.